clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6826,6 +6826,10 @@ def warn_floatingpoint_eq : Warning<
   "comparing floating point with == or != is unsafe">,
   InGroup<DiagGroup<"float-equal">>, DefaultIgnore;
 
+def warn_fenv_access : Warning<
+  "floating point environment access without #pragma STDC FENV_ACCESS set ON">,
+  InGroup<DiagGroup<"float-env">>;
```

Comment:
this should be a separate patch (and ideally would cover the range of standard fenv access functions)

Reply:
There's no immediate users until this patch as far as I'm aware. Unless you're talking about expanding it to cover some other cases I'm not aware of. I guess I could split out the clang / llvm stuff but this was generally more convenient.

Reply:
This should cover every function in fenv access. This low level implementation detail of those functions is less interesting to report on than the actual user facing functions

Reply:
I'm not specifically knowledgeable about how this should be handled in the grander scheme. Could I get rid of the warning and leave that for a separate patch?

Reply:
It must be a separate patch